hcl/hcl.go

```diff
@@ -977,239 +991,181 @@ func filterBlocksByType(blocktype string, blocks []*hclsyntax.Block) []*hclsynta
-				}
-
-			}
+			// TODO(katcipis): generate_file must be part of the whole Config.
+			// ignoring the block for now.
 		}
+	}
 
-		if !foundstack {
-			continue
+	tmBlock, ok := p.MergedBlocks["terramate"]
+	if ok {
+		var tmconfig Terramate
+		tmconfig, err := parseTerramateBlock(tmBlock)
+		if err != nil {
+			errs.Append(err)
+		} else {
+			config.Terramate = &tmconfig
 		}
+	}
+
+	globalsBlock, ok := p.MergedBlocks["globals"]
+	if ok {
+		errs.Append(globalsBlock.validateSubBlocks())
+
+		// value ignored in the main parser.
+	}
 
+	if foundstack {
 		logger.Debug().Msg("Parsing stack cfg.")
 
-		if tmconfig.Stack != nil {
+		if config.Stack != nil {
 			errs.Append(errors.E(errKind, stackblock.DefRange(),
 				"duplicated stack blocks across configs"))
 		}
 
-		tmconfig.Stack = &Stack{}
-		err := parseStack(tmconfig.Stack, stackblock)
+		config.Stack = &Stack{}
+		err := parseStack(config.Stack, stackblock)
 		if err != nil {
 			errs.Append(errors.E(errKind, err))
 		}
 	}
 
-	errs.Append(validateRunEnv(tmconfig))
-
 	if err := errs.AsError(); err != nil {
 		return Config{}, err
 	}
 
-	return tmconfig, nil
+	return config, nil
 }
 
-func validateRunEnv(config Config) error {
-	if config.Terramate == nil ||
-		config.Terramate.Config == nil ||
-		config.Terramate.Config.Run == nil ||
-		config.Terramate.Config.Run.Env == nil {
-		return nil
-	}
+func parseTerramateBlock(block *MergedBlock) (Terramate, error) {
+	logger := log.With().
+		Str("action", "parseTerramateBlock").
+		Logger()
 
+	logger.Trace().Msg("Range over terramate block attributes.")
+
+	tm := Terramate{}
+
+	errKind := ErrTerramateSchema
 	errs := errors.L()
-	attrs := map[string]Attribute{}
+	for _, attr := range block.Attributes.SortedList() {
+		value, diags := attr.Expr.Value(nil)
+		if diags.HasErrors() {
+			errs.Append(errors.E(errKind, diags))
+		}
+		switch attr.Name {
+		case "required_version":
+			logger.Trace().Msg("Parsing  attribute 'required_version'.")
 
-	for _, attr := range config.Terramate.Config.Run.Env.Attributes {
-		name := attr.Value().Name
-		if _, ok := attrs[name]; ok {
-			errs.Append(errors.E(
-				ErrTerramateSchema,
-				"redefined terramate.config.run.env attribute",
-				attr.Value().Range()))
+			if value.Type() != cty.String {
+				errs.Append(errors.E(errKind, attr.Expr.Range(),
+					"attribute is not a string"))
+
+				continue
+			}
+			if tm.RequiredVersion != "" {
+				errs.Append(errors.E(errKind, attr.NameRange,
+					"duplicated attribute"))
+			}
+			tm.RequiredVersion = value.AsString()
+
+		default:
+			errs.Append(errors.E(errKind, attr.NameRange,
+				"unsupported attribute"))
 		}
-		attrs[name] = attr
 	}
 
-	return errs.AsError()
+	errs.Append(block.validateSubBlocks("config"))
+
+	logger.Trace().Msg("Parse terramate sub blocks")
+
+	configBlock, ok := block.SubBlocks["config"]
+	if ok {
+		logger.Trace().Msg("Found config block.")
+
+		if tm.Config == nil {
+			tm.Config = &RootConfig{}
+		}
```

Comment:
```suggestion
		tm.Config = &RootConfig{}
```

If I'm not mistaken now that the blocks are already merged there is no need to keep checking if the field is nil (it won't be set multiple times anyway). Ignore if I missed something here.